scadalts-ui/src/components/amcharts/LineChart.js

```diff
@@ -0,0 +1,103 @@
+import * as am4core from '@amcharts/amcharts4/core';
+import * as am4charts from '@amcharts/amcharts4/charts';
+import am4themes_animated from '@amcharts/amcharts4/themes/animated';
+import httpClient from 'axios';
+import BaseChart from './BaseChart';
+
+am4core.useTheme(am4themes_animated);
+export class LineChart extends BaseChart {
+	constructor(chartReference, color, domain = '.') {
+		super(chartReference, 'XYChart', color, domain);
+	}
+	displayControls(scrollbarX, scrollbarY, legend) {
+		if (scrollbarX !== undefined && scrollbarX == 'true') {
+			this.showScrollbarX = true;
+		} else if (scrollbarX !== undefined && scrollbarX == 'false') {
+			this.showScrollbarX = false;
+		}
+		if (scrollbarY !== undefined && scrollbarY == 'true') {
+			this.showScrollbarY = true;
+		} else if (scrollbarY !== undefined && scrollbarY == 'false') {
+			this.showScrollbarY = false;
+		}
+		if (legend !== undefined && legend == 'true') {
+			this.showLegend = true;
+		} else if (legend !== undefined && legend == 'false') {
+			this.showLegend = false;
+		}
+	}
+	loadData(pointId, startTimestamp, endTimestamp, exportId) {
+		return new Promise((resolve, reject) => {
+			super.loadData(pointId, startTimestamp, endTimestamp, exportId).then((data) => {
+				if (this.pointCurrentValue.get(pointId) == undefined) {
+					this.pointCurrentValue.set(pointId, {
+						name: data.name,
+						suffix: data.textRenderer.suffix,
+						type: data.type,
+						labels: new Map(),
+					});
+				}
+				if (data.type === 'Multistate') {
+					let customLabels = data.textRenderer.multistateValues;
+					if (
+						data.textRenderer.typeName === 'textRendererMultistate' &&
+						customLabels !== undefined
+					) {
+						let labelsMap = new Map();
+						for (let i = 0; i < customLabels.length; i++) {
+							labelsMap.set(String(customLabels[i].key), customLabels[i].text);
+						}
+						this.pointCurrentValue.get(pointId).labels = labelsMap;
+					}
+				}
+				if (data.type === 'Binary') {
+					if (data.textRenderer.typeName === 'textRendererBinary') {
```

Comment:
//use const at the top of the file to compare types always it's possible
const TEXT_RENDERER_MULTISTATE = "textRendererMultistate";